### PR TITLE
create packages for plugins

### PR DESCRIPTION
**Pull request by dimkouv**
Extras:
    - un-export several entities
    - code style improvements
    - decouple some plugin libs (some func is declared under one plugin but used in another)
    - re-order methods / functions
    - delete unused vars

### CODE REVIEW

File: core/services/ocr2/plugins/ccip/executionplugin/factory.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
+package executionplugin
```
Comment (mateusz-sekara) [inline core/services/ocr2/plugins/ccip/executionplugin/factory.go:1]: I think execplugin is also okay and a bit shorter :D

Comment (mateusz-sekara) [inline core/services/ocr2/plugins/ccip/executionplugin/factory.go:1]: How about just `exec` and `commit`? I think there is no need to add this suffix

File: core/services/ocr2/plugins/ccip/commitplugin/plugin.go [hunk 2]
```diff
@@ -31,12 +30,59 @@ import (
 	"github.com/smartcontractkit/chainlink/v2/core/services/pipeline"
 )
 
-type BackfillArgs struct {
-	sourceLP, destLP                 logpoller.LogPoller
-	sourceStartBlock, destStartBlock int64
+const Label = "commit"
```
Comment (mateusz-sekara) [inline core/services/ocr2/plugins/ccip/commitplugin/plugin.go:33]: We don't have to export it, do we? Also I would make the name more meaningful, `Label` could be anything tbh. Maybe `pluginName` or `metricsPluginLabel` depends on where it's actually used

File: core/services/ocr2/plugins/ccip/commitplugin/plugin.go [hunk 3]
```diff
@@ -31,12 +30,59 @@ import (
...
+	wrappedPluginFactory := newFactory(*pluginConfig)
+
+	argsNoPlugin.ReportingPluginFactory = promwrapper.NewPromFactory(wrappedPluginFactory, "CCIPCommit", jb.OCR2OracleSpec.Relay, pluginConfig.destChainEVMID)
+	argsNoPlugin.Logger = relaylogger.NewOCRWrapper(pluginConfig.lggr, true, logError)
+	oracle, err := libocr2.NewOracle(argsNoPlugin)
+	if err != nil {
+		return nil, err
+	}
+	// If this is a brand-new job, then we make use of the start blocks. If not then we're rebooting and log poller will pick up where we left off.
+	if new {
```
Comment (mateusz-sekara) [inline core/services/ocr2/plugins/ccip/commitplugin/plugin.go:49]: I know it's not related but I would optimize for shorter block and flip the order here
```go
if !new {
     	return []job.ServiceCtx{job.NewServiceAdapter(oracle)}, nil
}

return ....
```

File: core/services/ocr2/plugins/ccip/executionplugin/plugin.go [hunk 4]
```diff
@@ -32,8 +32,96 @@ import (
 	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
 )
 
+const Label = "exec"
```
Comment (mateusz-sekara) [inline core/services/ocr2/plugins/ccip/executionplugin/plugin.go:35]: Similar to what I have mentioned for commitplugin

File: core/services/ocr2/plugins/ccip/internal/oraclelib/backfilled_oracle.go [hunk 5]
```diff
@@ -13,6 +13,11 @@ import (
 	"github.com/smartcontractkit/chainlink/v2/core/services/job"
 )
 
+type BackfillArgs struct {
```
Comment (mateusz-sekara) [inline core/services/ocr2/plugins/ccip/internal/oraclelib/backfilled_oracle.go:16]: Out of curiosity, why do need this struct and why is it named `BackfillArgs`.  It seems that we pass only logPoller instances there and starting blocks which are usually the blocks we deployed contracts.

**Comment by dimkouv** [comment]
Too many conflicts after the 'versioning abstraction' changes, will open another PR and start from scratch later

**Pull request closed without merging**